src/domain/usecases/GetIndicatorNumDenIDsUseCase.ts

```diff
@@ -0,0 +1,140 @@
+                    numDataElementsList: numerator.dataElementsList,
+                    numProgramDataElementsList: numerator.programDataElementsList,
+                    numCOCombosList: numerator.coCombosList,
+                    numPIndicatorsList: numerator.pIndicatorsList,
+                    numDataSetsList: numerator.dataSetsList,
+                    numProgramList: numerator.programList,
+                    denominator: trimLineBreaks(indicatorsItem.denominator),
+                    denDescription: indicatorsItem.denominatorDescription,
+                    denDataElementsList: denominator.dataElementsList,
+                    denProgramDataElementsList: denominator.programDataElementsList,
+                    denCOCombosList: denominator.coCombosList,
+                    denPIndicatorsList: denominator.pIndicatorsList,
+                    denDataSetsList: denominator.dataSetsList,
+                    denProgramList: denominator.programList,
+                };
+            })
+        );
+
+        return indicatorsData;
+    }
+}
+
+function trimLineBreaks(string: string) {
+    return string.replace(/\n|\r/g, "");
+}
+
+function trimItemSeparators(string: string, type: string) {
+    return string.replace(`${type}{`, "").replace("}", "");
+}
+
+function processDEParams(exp: string) {
+    const dataElems = _.uniq(exp.match(/(?:#\{.{11}\})/g)) ?? [];
+
+    return _(dataElems)
+        .flatMap(item => {
+            const dataElement = trimItemSeparators(item, "#");
+            return dataElement;
+        })
+        .uniq()
+        .value();
+}
+
+function processPDEParams(exp: string) {
+    const progDEs = exp.match(/(?:D\{.{11}\..{11}\})/g) ?? [];
+
+    return progDEs.flatMap(item => {
+        const cleanItem = trimItemSeparators(item, "D");
+        const [program, dataElement] = cleanItem.split(".");
+        return program && dataElement ? { dataElement: dataElement, program: program } : [];
+    });
+}
+
+function processCOCombosParams(exp: string) {
+    const coCombos = exp.match(/(?:#\{.{11}\..{11}\})/g) ?? [];
+
+    return coCombos.flatMap(item => {
+        const cleanItem = trimItemSeparators(item, "#");
+        const [dataElement, coCombo] = cleanItem.split(".");
+        return dataElement && coCombo ? { dataElement: dataElement, coCombo: coCombo } : [];
+    });
+}
+
+function processPIndicatorsParams(exp: string) {
+    const indicators = exp.match(/(?:I\{.{11}\})/g) ?? [];
+
+    return _(indicators)
+        .flatMap(item => {
+            const dataElement = trimItemSeparators(item, "I");
+            return dataElement;
+        })
+        .uniq()
+        .value();
+}
+
+function processDataSets(dataSets: Ref[], dataSetFilterList?: string[]) {
+    return _(dataSets)
+        .flatMap(dataSet =>
+            dataSetFilterList ? (dataSetFilterList.includes(dataSet.id) ? dataSet.id : []) : dataSet.id
+        )
+        .compact()
```

Comment:
You use this pattern, but I think you want map (which returns value or undefined), and then compact removed the undefined's

Reply:
Changed to:
```
function processDataSets(dataSets: Ref[], dataSetFilterList?: string[]) {
    return _(dataSets)
        .map(dataSet => {
            let id: string | undefined = undefined;

            if ((dataSetFilterList && dataSetFilterList.includes(dataSet.id)) || !dataSetFilterList) {
                id = dataSet.id;
            }

            return id;
        })
        .compact()
        .uniq()
        .value();
```